### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by Nashtare**
# Description

This PR refactors tests from the key_generation module to reduce code duplication. It particular:

- it moves the `do_keygen` helper method for testing from `signature` to `key_generation` and adds some additional sanity checks within
- reuses it for _regular_ scenarios testing (i.e. no failures)
- adds two edge cases checks (i.e. resharing from a single party and to a single party)

### CODE REVIEW

File: src/dkg/key_generation.rs [hunk 1]
```diff
@@ -1196,6 +1196,197 @@ mod test {
...
+            Option<ThresholdParameters<Secp256k1Sha256>>,
+            Option<Vec<IndividualSigningKey<Secp256k1Sha256>>>,
+        ),
+    > {
+        type Dkg<T> = DistributedKeyGeneration<T, Secp256k1Sha256>;
+
+        let params = ThresholdParameters::new(n1, t1);
+        let rng = OsRng;
+
+        let mut participants = Vec::<Participant<Secp256k1Sha256>>::new();
```
Comment (hratoanina) [inline src/dkg/key_generation.rs:1219]: Nit: We can use `with_capacity` (similarly for the other `new`s in the function).

**Review by hratoanina — APPROVED** [review]
Looks good, thanks!

File: src/dkg/key_generation.rs (hunk 1 above)
Comment (Nashtare) [reply src/dkg/key_generation.rs:1219]: Yeah, though I've never bothered with these in the past as they are within tests only
